python/lsst/cp/pipe/ptc/cpSolvePtcTask.py

```diff
@@ -416,20 +431,27 @@ def _getInitialGoodPoints(means, variances, minVarPivotSearch):
 
         Notes
         -----
-        Eliminate points beyond which the variance decreases
+        Eliminate points beyond which the variance decreases.
         """
         goodPoints = np.ones_like(means, dtype=bool)
         pivotList = np.where(np.array(np.diff(variances)) < 0)[0]
         if len(pivotList) > 0:
             # For small values, sometimes the variance decreases slightly
             # Only look when var > self.config.minVarPivotSearch
             pivotList = [p for p in pivotList if variances[p] > minVarPivotSearch]
+            # Require that the varince decreases during
+            # consecutivePointsVarDecreases
+            # consecutive points. This will give a first
+            # estimate of the PTC turn-off, which
+            # may be updated (reduced) further in the code.
             if len(pivotList) > 1:
-                # Require that the decrease in variance happen for two
-                # consecutive signal levels
-                pivotIndex = np.min(np.where(np.diff(pivotList) == 1)[0])
-                pivot = pivotList[pivotIndex]
-                goodPoints[pivot+1:] = False
+                for k, g in groupby(enumerate(pivotList), lambda x: x[0]-x[1]):
```

Comment:
Can this have a quick comment describing what it's doing?  I think I follow along, but even just a per-line  `# Find all indices in this group` kind of thing would clarify what's happening.